src/main/scala/gitbucket/core/controller/AccountController.scala

```diff
@@ -150,20 +150,21 @@ trait AccountControllerBase extends AccountManagementControllerBase {
 
   get("/:userName/_avatar"){
     val userName = params("userName")
+    contentType = "image/png"
     getAccountByUserName(userName).map{ account =>
       response.setDateHeader("Last-Modified", account.updatedDate.getTime)
       account.image.map{ image =>
         RawData(FileUtil.getMimeType(image), new java.io.File(getUserUploadDir(userName), image))
       }.getOrElse{
-        contentType = "image/png"
         (if (account.isGroupAccount) {
           TextAvatarUtil.textGroupAvatar(account.fullName)
         } else {
           TextAvatarUtil.textAvatar(account.fullName)
-        }).getOrElse(Thread.currentThread.getContextClassLoader.getResourceAsStream("noimage.png"))
+        }).get
```

Comment:
For example, maybe you can write as following in this case:

```scala
getAccountByUserName(userName).flatMap { account =>
  response.setDateHeader("Last-Modified", account.updatedDate.getTime)
  account.image.map { image =>
    Some(RawData(FileUtil.getMimeType(image), new java.io.File(getUserUploadDir(userName), image)))
  }.getOrElse {
    if (account.isGroupAccount) {
      TextAvatarUtil.textGroupAvatar(account.fullName)
    } else {
      TextAvatarUtil.textAvatar(account.fullName)
    }
  }
}.getOrElse {
  response.setHeader("Cache-Control", "max-age=3600")
  Thread.currentThread.getContextClassLoader.getResourceAsStream("noimage.png")
}
```

But I wrote this code in my brain, so I don't know whether this code works fine :-P
Could you try it?